src/extension.rs

```diff
@@ -123,17 +120,28 @@ fn split_extension(name: &mut &[u8]) -> Option<Extension> {
     Some(ext)
 }
 
-pub fn parse_format(fmt: &OsStr) -> crate::Result<Vec<Extension>> {
-    let fmt = <[u8] as ByteSlice>::from_os_str(fmt).ok_or_else(|| Error::InvalidFormat {
-        reason: "Invalid UTF-8".into(),
+pub fn parse_format_flag(input: &OsStr) -> crate::Result<Vec<Extension>> {
```

Comment:
I think we can just strip the leading `.` here?

Reply:
We could, but I already did this code that makes it even "more forgiving", while unrealistic that someone will type ".tar..gz" I also don't see a problem with it so I'll keep it.